### ISSUE TITLE
Issue with http://icanhazip.com/ being out of service

### ISSUE THREAD

**Issue by nharrisanalyst** [chrisbolin/react-detect-offline#41]
This morning the website http://icanhazip.com/ was sending 503 errors meaning that it was wrongfully giving my application an error for being offline.

as this module depends on http://ipv4.icanhazip.com/  to detect you isp and if you are online so   is their anyway we can modify this so that if http://ipv4.icanhazip.com/ is sending 503 errors that we do not give a false positive when trying to detect if one is offline?

**Comment by nharrisanalyst** [comment]
or if anyone has any suggestions on work i good do to fix this

**Comment by chrisbolin** [comment]
@nharrisanalyst thanks for the heads up! if you want to fix this, i'd love a PR that considered _any_ response from the server as an "online" signal. e.g. 200, 301, 404, 503. All of those should signal that the application is online.

**Comment by nharrisanalyst** [comment]
https://twitter.com/majorhayden/status/1126859130268987393

**Comment by chrisbolin** [comment]
this is fully configurable, too: https://github.com/chrisbolin/react-detect-offline#props

just change `polling.url` to something you have more confidence in :D

**Comment by kpereda** [comment]
So the idea would be to just verify that the response exists and remove the `xhr.status >= 200 && xhr.status < 400` verification?

```
if (xhr.readyState === xhr.HEADERS_RECEIVED) {
        if (xhr.status) {
          isOnline();
        } else {
          isOffline();
        }
      }
```

> @nharrisanalyst thanks for the heads up! if you want to fix this, i'd love a PR that considered _any_ response from the server as an "online" signal. e.g. 200, 301, 404, 503. All of those should signal that the application is online.

**Comment by chrisbolin** [comment]
Exactly!

**Comment by chrisbolin** [comment]
testing @kpereda's fix right now. looks good so far!

**Comment by chrisbolin** [comment]
@kpereda thanks for the quick fix!

**Comment by chrisbolin** [comment]
@nharrisanalyst thanks for reporting

this is fixed in [**react-detect-offline@2.4.0**](https://www.npmjs.com/package/react-detect-offline)
